common/dynamicconfig/constants.go

```diff
@@ -1520,6 +1520,12 @@ const (
 	// Default value: true
 	// Allowed filters: DomainName
 	EnableReadVisibilityFromPinot
+	// EnableReadVisibilityFromOS is key for enable read from opensearch or db visibility, usually using with AdvancedVisibilityWritingMode for seamless migration from db visibility to advanced visibility
+	// KeyName: system.enableReadVisibilityFromOS
+	// Value type: Bool
+	// Default value: true
+	// Allowed filters: DomainName
+	EnableReadVisibilityFromOS
```

Comment:
There are 3 bool flags which can be set independently. What if more than one is true or all is false? Behavior is not clear from config perspective. Ideally there should be one flag `ReadVisibilityStoreName`

Reply:
Yeah, that can help preventing adding more flags here. When advanced visibility is enabled, it will first read from the advanced visibility store (ES or OS/Pinot), ES always has the most priority. If all of them are false, it will fall back to read from DB. 
https://github.com/cadence-workflow/cadence/blob/master/common/persistence/visibility_dual_manager.go#L318

Reply:
https://github.com/cadence-workflow/cadence/blob/master/common/persistence/visibility_triple_manager.go#L666
the triple manager has more cases

Reply:
Adding a new dynamicconfig to all the environments is also a lot of manual work. If it's feasible to switch to a single flag let's do that